openstf/model/objective.py

```diff
@@ -55,6 +119,11 @@ def __call__(self, trial: optuna.trial.FrozenTrial) -> float:
         Returns:
             float: Mean absolute error for this trial.
         """
+        # Check elapsed time
+        elapsed = datetime.utcnow() - self.start_time
+        if elapsed > timedelta(minutes=2, seconds=0):
+            trial.study.stop()
```

Comment:
Very useful functionality! Minor remarks:
For maintainability; please define `MAX_DURATION_MINUTES = 2` at the top of the file, or even better: as a optional function argument, and refer to that variable here. You would have to turn the 'elapsed' then into something with minutes by: `elapsed_minutes=elapsed.total_seconds()/60.`
Also, would be nice to include a logging step if the study is stopped due to time constraint 
`logging.info('Study stopped due to time constraint')`
Suggestion instead of 'elapsed': `study_duration`
Please include this time constraint in the docstring